Library/Formula/fb-infer.rb

```diff
@@ -0,0 +1,53 @@
+class FbInfer < Formula
+  desc "Static analyzer for Java, C and Objective-C"
+  homepage "http://fbinfer.com/"
+  url "https://github.com/facebook/infer/releases/download/v0.5.0/infer-osx-v0.5.0.tar.xz"
+  sha256 "6a8547ac0b75a5e2bbeccae2169e39f753a60adbcacb6c94599fd31343a71ce7"
+  head "https://github.com/facebook/infer.git"
+
+  option "with-clang-only", "Installs Infer with only Clang support."
+  option "with-java-only", "Installs Infer with only Java support."
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "opam" => :build
+
+  def install
+    target_language = "all"
+    if build.with?("clang-only") && build.with?("java-only")
+      odie "Options --with-clang-only and --with-java-only are mutually exclusive."
+    end
+
+    if build.with?("java-only")
+      target_language = "java"
+    elsif build.with?("clang-only")
+      target_language = "clang"
+    end
+
+    opamroot = buildpath/"build"
+    opamroot.mkpath
+    ENV["OPAMROOT"] = opamroot
+    ENV["OPAMYES"] = "1"
+
+    system "opam", "init", "--no-setup"
+    system "opam", "update"
+
+    system "opam", "install", "ocamlfind"
+    system "opam", "install", "sawja>=1.5.1"
+    system "opam", "install", "atdgen>=1.6.0"
+    system "opam", "install", "extlib>=1.5.4"
+
+    system "./build-infer.sh", target_language, "--yes"
+
+    libexec.install "infer/annotations"
+    libexec.install "infer/lib"
+    libexec.install "infer/bin"
+
+    bin.install_symlink libexec/"bin/infer"
+  end
+
+  test do
+    system "#{bin}/infer"
```

Comment:
I suspect this is just printing usage instructions. Can you make this test do something more substantial than this? A good test case would be something that uses the actual functionality of `infer`, so probably a minimal Java and/or C program that causes `infer` to report an error.